flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/source/TestIcebergSourceContinuous.java

```diff
@@ -58,9 +61,11 @@
 
 public class TestIcebergSourceContinuous {
 
+  public static final InMemoryReporter METRIC_REPORTER = InMemoryReporter.create();
+
   @ClassRule
   public static final MiniClusterWithClientResource MINI_CLUSTER_RESOURCE =
-      MiniClusterResource.createWithClassloaderCheckDisabled();
+      MiniClusterResource.createWithClassloaderCheckDisabled(METRIC_REPORTER);
```

Comment:
Could we use junit5 tests for testing?

Reply:
That would require refactoring this whole class. I can refactor it but that is outside the scope of this PR. Otherwise, separate out the metric testing to a new class and use junit5 there.

Reply:
How about moving the existing test to junit5 in another PR, and rebasing this one above that one.

@nastra: Is there any ongoing work to move this test to junit5?

Reply:
I haven't seen a PR that would include migrating this class here to JUnit5

Reply:
@pvary @stevenzwu @nastra I'm fine with that and I can volunteer to migrate this class to JUnit5 after the PR is merged. I am limited in my bandwidth this week, so I can address it next week, while I would like these metrics to land in the upcoming iceberg release.

Reply:
I agree with @mas-chen that the JUnit5 can be tackled as a separate PR. this PR doesn't add any new test classes.

Reply:
@mas-chen: It seems more complicated this way (modify, migrate, remove), but I am mostly interested in the final result 😀